### PR TITLE
feature/AT-7363-file-upload-base-component

### PR DESCRIPTION
**Pull request by tonysumner-ccpo**

### CODE REVIEW

File: src/sass/components/ATATFileUpload.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.atat-file-upload,
+.atat-file-upload.v-input--is-focused.primary--text{
+    background-image: url("data:image/svg+xml,%3csvg width='100%25' height='100%25' xmlns='http://www.w3.org/2000/svg'%3e%3crect width='100%25'  height='100%25' fill='%23fbfbfb' stroke='%23544496' stroke-width='2' stroke-dasharray='10' stroke-dashoffset='2' stroke-linecap='square'/%3e%3c/svg%3e") !important;
```
Comment (ericyoungquist-ccpo) [inline src/sass/components/ATATFileUpload.scss:3]: pls change `background-image` to 
```
url("data:image/svg+xml,%3csvg width='100%25' height='100%25' xmlns='http://www.w3.org/2000/svg'%3e%3crect width='100%25' height='100%25' fill='none' rx='8' ry='8' stroke='%2361686c' stroke-width='2' stroke-dasharray='8%2c 8' stroke-dashoffset='0' stroke-linecap='square'/%3e%3c/svg%3e") !important;
```

File: src/components/ATATFileUpload.vue [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      accept="application/pdf,application/vnd.ms-excel"
+      :truncate-length="truncateLength"
+      :clearable="false"
+      @change="fileUploadChanged"
+    >
+      <template v-slot:prepend-inner>
+        <div class="content text-center">
+          <v-icon class="icon-60 mt-10 mb-5">upload_file</v-icon>
+          <h2>Drag and Drop</h2>
+          <p class="mt-1 mb-4 d-flex justify-center">your file here or
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:21]: pls change "your file here or" text color to base darkest `.bg-base-darkest`

File: src/components/ATATFileUpload.vue [hunk 3]
```diff
@@ -0,0 +1,152 @@
+<template>
+  <div v-cloak
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:2]: General layout thoughts: Using the attached, Figma on left, SPA on right, some minor margins/paddings adjustments should be made to match Figma more precisely.
Screen Shot 2022-05-04 at 6 22 45 PM

File: src/components/ATATFileUpload.vue [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      <template v-slot:prepend-inner>
+        <div class="content text-center">
+          <v-icon class="icon-60 mt-10 mb-5">upload_file</v-icon>
+          <h2>Drag and Drop</h2>
+          <p class="mt-1 mb-4 d-flex justify-center">your file here or 
+            <v-btn 
+              class="link-button inline-text" 
+              @mousedown="fileUploadClicked">
+                browse to upload
+              </v-btn>
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:26]: pls change from `v-btn` to `a` - unusual hover state. Also add ID for testing
```            
            <a
              role="button"
              id="BrowseToUpload"
              class="_text-link ml-1" 
              @mousedown="fileUploadClicked"
            >
                browse to upload
            </a>
```

File: src/components/ATATFileUpload.vue [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+  /**
+   * removes files with extensions not defined in 
+   * this.validFileFormats property
+   * 
+   * @params (files: FileList)
+   * 
+   */
+  private removeInvalidFiles(files: FileList): void {
+    debugger;
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:134]: remove debugger

File: src/components/ATATFileUpload.vue [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    <h2 class="mt-5" v-if="uploadedFiles.length>0">Files Uploaded</h2>
+    <ul v-for="(file, idx) in uploadedFiles" :key="idx">
+      <li>{{ file.name }}</li>
+    </ul>
+   </div>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { Component, Prop, PropSync } from "vue-property-decorator";
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:42]: remove unused PropSync?

File: src/components/ATATFileUpload.vue [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    };
+  }
+
+  // props
+  @Prop({default: 15}) private truncateLength!: string;
+  @Prop({default: ""}) private id!: string;
+  @Prop({default: ()=>[]}) private validFileFormats!: string[];
+
+  //data
+  // @PropSync("files", {default: ()=>[]}) private _files: File[] = [];
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:62]: Remove commented code? or leaving for later?

File: src/sass/components/ATATButton.scss [hunk 8]
```diff
@@ -68,6 +68,11 @@
   &.link-button, &.tertirary {
     padding: 6px 12px !important;
 
+    &.inline-text{
+      padding: 0px 4px !important;
```
Comment (ericyoungquist-ccpo) [inline src/sass/components/ATATButton.scss:72]: class won't be necessary if changing `v-btn` to `a` tag

File: src/steps/10-FinancialDetails/FundingPlan.vue [hunk 9]
```diff
@@ -1,14 +1,21 @@
 
 <template>
   <div>
-    Future Funding Plan page
+    <ATATFileUpload 
+    :validFileFormats="['xlsx', 'xls','pdf']"
+    id="FundingPlan"/>
```
Comment (ericyoungquist-ccpo) [inline src/steps/10-FinancialDetails/FundingPlan.vue:6]: indent lines 5 and 6, drop closing `/>` to be aligned with opening `<`tag

File: src/components/ATATFileUpload.vue [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+   * 2. removes unnecessary vuetify status msg
+   */
+  private fileUploadChanged(): void {
+    this.removeInvalidFiles(this.fileUploadControl.files as FileList)
+    Vue.nextTick(()=>{
+      //remove default vuetify status that displays after
+      //upload (eg. '2 files')
+      const vuetifyFileUploadStatus = 
+        document.getElementsByClassName("v-file-input__text")[0] as HTMLDivElement;
+      console.log(vuetifyFileUploadStatus);
```
Comment (ericyoungquist-ccpo) [inline src/components/ATATFileUpload.vue:86]: remove console log?

**Review by ericyoungquist-ccpo — CHANGES_REQUESTED** [review]

**Comment by ericyoungquist-ccpo** [comment]
Maybe not for this ticket, but we should soon look into adding Roboto Medium (weight 500) to our codebase. The "Drag and Drop" text is currently bold (700) in the SPA, but 500 in Figma.

File: src/components/ATATFileUpload.vue (hunk 7 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:62]: yes, leaving for later.

File: src/components/ATATFileUpload.vue (hunk 10 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:86]: oopsies

File: src/components/ATATFileUpload.vue (hunk 3 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:2]: - [x] aligning issues resolved. Figma on the left, web page on the right
image

File: src/components/ATATFileUpload.vue (hunk 4 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:26]: - [x] Done

File: src/components/ATATFileUpload.vue (hunk 5 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:134]: -[x] Removed

File: src/sass/components/ATATButton.scss (hunk 8 above)
Comment (tonysumner-ccpo) [reply src/sass/components/ATATButton.scss:72]: -[x] removed

File: src/steps/10-FinancialDetails/FundingPlan.vue (hunk 9 above)
Comment (tonysumner-ccpo) [reply src/steps/10-FinancialDetails/FundingPlan.vue:6]: - [x] Done

File: src/components/ATATFileUpload.vue (hunk 6 above)
Comment (tonysumner-ccpo) [reply src/components/ATATFileUpload.vue:42]: - [ ] Removed

File: src/sass/components/ATATFileUpload.scss (hunk 1 above)
Comment (tonysumner-ccpo) [reply src/sass/components/ATATFileUpload.scss:3]: - [ ] Done, except for the fill color.  Leaving that in b/c on the FIGMA is offwhite or #fbfbfb.
image

Comment (ericyoungquist-ccpo) [reply src/sass/components/ATATFileUpload.scss]: ah yes, I was focusing only on the border - forgot the background color

File: src/components/ATATFileUpload.vue (hunk 3 above)
Comment (ericyoungquist-ccpo) [reply src/components/ATATFileUpload.vue:2]: looks good - I was just pointing out general spacing issues. this is much better!

File: src/components/ATATFileUpload.vue (hunk 4 above)
Comment (ericyoungquist-ccpo) [reply src/components/ATATFileUpload.vue]: correct - no `href` if `role="button"`

**Review by ericyoungquist-ccpo — APPROVED** [review]

**Pull request merged**